js/setup.js

```diff
@@ -77,23 +90,102 @@ function getHtmlWizard(wizard) {
   return elem;
 }
 
-function getHtmlWizards(wizards) {
+function getHtmlWizardsFragment(wizards) {
   const fragment = document.createDocumentFragment();
 
   wizards.forEach((wizard) => {
-    fragment.appendChild(getHtmlWizard(wizard));
+    fragment.appendChild(getHtmlWizardElement(wizard));
   });
 
   return fragment;
 }
 
 function initModal() {
-  const wizards = generateWizards(4);
-  const htmlFragment = getHtmlWizards(wizards);
+  const wizards = generateWizardsDataArr(WIZARDS_COUNT);
+  const htmlFragment = getHtmlWizardsFragment(wizards);
+
+  modalElementsObj.wizards.list.appendChild(htmlFragment);
+  modalElementsObj.wizards.container.classList.remove(`hidden`);
+}
+
+function modalEscKeydownHandler(evt) {
+  if (window.isEscButton(evt.key) && document.activeElement !== modalElementsObj.nameInput) {
+    evt.preventDefault();
+    closeModal();
+  }
+}
 
-  modalElements.wizards.list.appendChild(htmlFragment);
-  modalElements.wizards.container.classList.remove(`hidden`);
-  modal.classList.remove(`hidden`);
+function openModal() {
+  modalNode.classList.remove(`hidden`);
+  document.addEventListener(`keydown`, modalEscKeydownHandler);
 }
 
+function closeModal() {
+  modalNode.classList.add(`hidden`);
+  document.removeEventListener(`keydown`, modalEscKeydownHandler);
+}
+
+openButtonNode.addEventListener(`click`, (evt) => {
+  evt.preventDefault();
+  openModal();
+});
```

Comment:
`evt.preventDefault()` = предотвращение действия по умолчанию. 

Это применимо в основном:
- на ссылках, чтобы по срабатыванию события не происходил переход по ссылке
- на кнопках с типом `submit` внутри форм, чтобы по срабатыванию события не отправлялась форма

Тут у тебя происходит клик по обычной кнопке, у нее и так нет действия по умолчанию, предотвращать нечего. Проверь остальные случаи использования `preventDefault` в этом файле, кажется от каждого из них можно избавиться

```suggestion
openButtonNode.addEventListener(`click`, openModal);
```

Аналогично с кликом закрытия